src/extension_field_gt.rs

```diff
@@ -51,6 +58,20 @@ impl GT {
         Self { value: fexp(&e) }
     }
 
+    /// Reduced ate double pairing which inverts the first point so unity can be checked
+    pub fn ate_2_pairing_cmp(g1: &G1, g2: &G2, h1: &G1, h2: &G2) -> bool {
+        if g1.is_identity() || g2.is_identity() {
+            return Self::ate_pairing(h1, h2).is_one();
+        }
+        if h1.is_identity() || h2.is_identity() {
+            return Self::ate_pairing(g1, g2).is_one();
+        }
+
+        let p = g1.negation();
```

Comment:
I can say after some experience of using `GT` that you should not decide which element needs to be negated as the caller might already have some element negated from previous computations